### ISSUE TITLE
CoaL crash during GCC install left GCC uninstallable/unremovable

### ISSUE THREAD

**Issue by russelldb** [joyent/pkgin#4]
Sorry for the vague report.

I was installing GCC49 on (I think) `c193a558-1d63-11e9-97cf-97bb3ee5c14f  base-64-lts                     18.4.0                                      smartos  zone-dataset  2019-01-21`
when my vmware just halted the CoaL VM without any warning, and when I finally got back in, gcc was not installed, not installable, not removable. I destroyed the zone and started over. 

Again, sorry I didn't think to gather more info at the time.

**Comment by rmustacc** [comment]
On 2/20/19 5:52 , Russell Brown wrote:
> Sorry for the vague report.
> 
> I was installing GCC49 on (I think) `c193a558-1d63-11e9-97cf-97bb3ee5c14f  base-64-lts                     18.4.0                                      smartos  zone-dataset  2019-01-21`
> when my vmware just halted the CoaL VM without any warning, and when I finally got back in, gcc was not installed, not installable, not removable. I destroyed the zone and started over. 
> 
> Again, sorry I didn't think to gather more info at the time.
> 

When you say that VMware halted it, did it indicate that a triple fault
occurred, did the operating system panic, or something else occur?

**Comment by jperkin** [comment]
I released 0.11.7 which changed the database pragmas to avoid unsafe operations, this should hopefully at least fix the problems where the pkgin database gets corrupted and other issues that Robert was also seeing.

There's still the issue where if you interrupt a package install, the pkg database will say the package is installed but not all the files are on disk.  This will require changes to pkg_install to fix, until then performing a "pkgin remove" followed by a "pkgin install" should be ok.

I'm going to close this one out for now, as I've transferred the repository back over to the parent NetBSDfr/pkgin, and there appears to be no way to transfer issues across organisations yet.  However the pkg_install issue is very much something I want to look at at some point.
